mail_activity_reminder/models/mail_activity.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2020 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime, time
+from dateutil.relativedelta import relativedelta
+from pytz import timezone, UTC
+
+from odoo import _, api, fields, models
+
+
+class MailActivity(models.Model):
+    _inherit = 'mail.activity'
+
+    next_reminder = fields.Datetime(
+        string='Next reminder',
+        compute='_compute_next_reminder',
+        compute_sudo=True,
+        store=True,
+    )
+    last_reminder_local = fields.Datetime(
+        string='Last reminder (local)',
+    )
+
+    @api.model
+    def _get_activities_to_remind_domain(self):
+        """Hook for extensions"""
+        return [('next_reminder', '<=', fields.Datetime.now())]
+
+    @api.model
+    def _get_activities_to_remind(self):
+        return self \
+            .search(self._get_activities_to_remind_domain()) \
+            .filtered(lambda activity: activity._should_be_reminded())
```

Comment:
here you will loop twice as well, because later in `action_remind` you loop already on each record. 
I'd move the call to `_should_be_reminded` inside the loop of `action_remind`.

Reply:
I'd prefer not doing that, to keep checks from actions a bit separate.

Reply:
as you prefer, but you loop twice for nothing (see my comment above on filtered).

Reply:
Yep, but values are going to be reused due to being prefetched, so I don't really get the harm of it.